### ISSUE TITLE
after 19.7 upgrade VIRTIO NICs not detected anymore

### ISSUE THREAD

**Issue by pmoser1976** [opnsense/core#3611]
I run OPNSense on UNRaid VM (Q35) using virtual NICs (VIRTIO).

On 19.1 only Q35-2.6 worked, using other Q35-x.x there were no NICs detected in OPNsense.
So I ran Q35-2.6, everything fine.

Since update to 19.7 there are no NICs detected anymore!! I tried other Q35 versions, also tried a fresh install an a new VM. => 19.1 shows NICs, 19.7 does not find NICs!!

Now I run a Backup on 19.1 again.

**Important notices**
Before you add a new report, we ask you kindly to acknowledge the following:

[-] I have read the contributing guide lines at https://github.com/opnsense/core/blob/master/CONTRIBUTING.md

[-] I have searched the existing issues and I'm convinced that mine is new.

**Describe the bug**

**To Reproduce**

**Expected behavior**

**Screenshots**

**Relevant log files**
If applicable, information from log files supporting your claim.

**Additional context**

**Environment**
Software version used and hardware type if relevant.
e.g.:

OPNsense 19.1.1 (amd64, OpenSSL).
Intel® Xeon™ E3-1225V5 3.3Ghz Quad Core
Network Intel® I210-AT

**Comment by AdSchellevis** [comment]
The kernel we use for 19.7 is quite similar to the one in the latest 19.1.

To pinpoint the issue, can you test the following and let us know when/if your issue appears:

- Install 19.1.4 (https://pkg.opnsense.org/releases/19.1/)
- Install new netmap kernel, run from a console : `opnsense-update -kr 19.1.4-netmap`
- Reboot

**Comment by pmoser1976** [comment]
I was able to Install 19.1.4 in VM (NICs okay)
I was able to update Kernel in same VM
After reboot I have 100% CPU-usage on host and VM does not start correctly 
![beforeRebootEverythingOkay](https://user-images.githubusercontent.com/24461961/62061244-1195a800-b227-11e9-8008-f318916e5a25.png)
![afterRebootWithNewKernel](https://user-images.githubusercontent.com/24461961/62061245-1195a800-b227-11e9-9f63-6140efdc27c8.png)

**Comment by AdSchellevis** [comment]
root mount issues can't really be related to the netmap kernel. No clue what the issue with your setup is, but it looks broader than a network driver issue.

**Comment by pmoser1976** [comment]
please tell me if I can test something else to explore the issue in more detail.
thank you.

**Comment by lordraiden** [comment]
I have exactly the same problem.
in 19.1 it was working, after the upgrade networks cards were not detected.
I am using q35 3.1 and the nics are passthough to the VM. I am using IOMMU

If it worth something I had the same issue with pfsense 2.5
It looks like is not an isolated issue
https://forum.opnsense.org/index.php?topic=13607.0

**Comment by tsalwierz** [comment]
This issue does happen with Q35 but not i440fx. I'm not sure if this is a opnsense issue at all. Have any of you opened a bug report with QEMU?

**Comment by lordraiden** [comment]
@pmoser1976 maybe you can post the bug here
https://bugs.launchpad.net/qemu/

I could do it but I think you are more tekkie than I do and I won't know how to anwer their questions.

I think it might be a problem between the FreeBSD version that opnsense 19.7 and pfsense 2.5 are running and Qemu.

**Comment by lordraiden** [comment]
> 
> 
> This issue does happen with Q35 but not i440fx. I'm not sure if this is a opnsense issue at all. Have any of you opened a bug report with QEMU?

i440fx isn't suppose to be for windows sysmtems? would it have a worse performance compared with Q35?

**Comment by tsalwierz** [comment]
Link me to where you heard i440fx is windows only.

**Comment by mr44er** [comment]
I don't know this unraid-thingie, but is it possible to switch Q35/i440fx to virtio/virtio-net?

I use bhyve on FreeBSD and HardenedBSD/OPNsense has these virtio-drivers built in.

**Comment by tsalwierz** [comment]
> I don't know this unraid-thingie, but is it possible to switch Q35/i440fx to virtio/virtio-net?
> It could need a fresh install of the vm, but I think it's worth the shot.

Yes. Enable virtio in tunables. You swap a setting in the QEMU config and done. No need to reinstall.

**Comment by lordraiden** [comment]
I think it might be related to this

Bug 236922 - Virtio fails as QEMU-KVM guest with Q35 chipset 
https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=236922

FreeBSD 12-RELEASE guest in QEMU/KVM
https://forums.freebsd.org/threads/freebsd-12-release-guest-in-qemu-kvm.70207/

**Comment by marshalleq** [comment]
In case it's relevant, I am also getting this error, but not only is the virtio not being detected, neither are the passthrough nics.

**Comment by fichtner** [comment]
This might be a FreeBSD 11.3 -ish related patch issue. We're still on 11.2 but with a few networking patches for better netmap support which landed in 11.3.
